ui/components/multichain/asset-picker-amount/asset-balance/asset-balance-text.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import UserPreferencedCurrencyDisplay from '../../../app/user-preferenced-currency-display';
+import { PRIMARY } from '../../../../helpers/constants/common';
+import TokenBalance from '../../../ui/token-balance';
+import { Asset } from '../../../../ducks/send';
+import {
+  getCurrentCurrency,
+  getSelectedAccountCachedBalance,
+} from '../../../../selectors';
+import { AssetType } from '../../../../../shared/constants/transaction';
+import {
+  type TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import CurrencyDisplay from '../../../ui/currency-display';
+import { useTokenTracker } from '../../../../hooks/useTokenTracker';
+import { useCurrencyDisplay } from '../../../../hooks/useCurrencyDisplay';
+import { useTokenFiatAmount } from '../../../../hooks/useTokenFiatAmount';
+import useIsFiatPrimary from '../hooks/useIsFiatPrimary';
+
+export interface AssetBalanceTextProps {
+  asset: Asset;
+  balanceColor: TextColor;
+}
+
+export function AssetBalanceText({
```

Comment:
can you write unit tests for this new component , here is my suggestion:
```
import React from 'react';
import { render } from '@testing-library/react';
import { AssetBalanceText } from './AssetBalanceText'; // Adjust the import path as necessary
import { AssetType } from '../../../../../shared/constants/transaction'; // Adjust import path
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // You might need to install redux-mock-store

// Mock hooks
jest.mock('../../../../hooks/useIsFiatPrimary', () => ({
  useIsFiatPrimary: jest.fn(),
}));

jest.mock('../../../../hooks/useTokenTracker', () => ({
  useTokenTracker: jest.fn(),
}));

jest.mock('../../../../hooks/useCurrencyDisplay', () => ({
  useCurrencyDisplay: jest.fn(),
}));

jest.mock('../../../../hooks/useTokenFiatAmount', () => ({
  useTokenFiatAmount: jest.fn(),
}));

const mockStore = configureStore([]);
const store = mockStore({});

describe('AssetBalanceText', () => {
  it('matches snapshot', () => {
    const asset = {
      type: AssetType.native,
      details: { address: '0x0', decimals: 18 },
      balance: '100',
    };
    const { asFragment } = render(
      <Provider store={store}>
        <AssetBalanceText asset={asset} balanceColor="textColor" />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders fiat primary correctly', () => {
    useIsFiatPrimary.mockImplementation(() => true);

    const { getByText } = render(
      <Provider store={store}>
        <AssetBalanceText asset={asset} balanceColor="textColor" />
      </Provider>
    );
    expect(getByText('Some fiat value')).toBeInTheDocument();
  });

  // Test for native asset type
  it('renders native asset type correctly', () => {
    const asset = {
      type: AssetType.native,
      details: { address: '0x0', decimals: 18 },
      balance: '100',
    };

    const { getByText } = render(
      <Provider store={store}>
        <AssetBalanceText asset={asset} balanceColor="textColor" />
      </Provider>
    );
  });
});
```

not sure how accurate is this , but i think you got the idea